sys/base64/base64.c

```diff
@@ -59,8 +59,7 @@ static char getsymbol(unsigned char code)
 int base64_encode(unsigned char *data_in, size_t data_in_size, \
                   unsigned char *base64_out, size_t *base64_out_size)
 {
-    size_t padding_bytes = ((data_in_size % 3) ? (3 - (data_in_size % 3)) : 0);
-    size_t required_size = (4 * (data_in_size + 2 - ((data_in_size + 2) % 3)) / 3) + padding_bytes;
+    size_t required_size = ((4 * data_in_size / 3) + 3) & ~3;
```

Comment:
Multiplying before dividing may introduce an integer overflow. What's wrong with the formula @x3ro posted? `4 * ((input_length + 2) / 3)`